### PR TITLE
[Evaluation]: Log openhands version in eval output folder, instead of agent version

### PR DESCRIPTION
**Pull request by xingyaoww**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

There's a lot of things that could impact agent quality since the start of this project - e.g, reliability of bash terminal, better implementation of memory condensation, etc. So I think it probably make sense to track these eval runs by OpenHands version, instead of agent version.

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:417b09f-nikolaik   --name openhands-app-417b09f   docker.all-hands.dev/all-hands-ai/openhands:417b09f
```

### CODE REVIEW

File: evaluation/utils/version_control.sh [hunk 1]
```diff
@@ -39,8 +39,8 @@ checkout_original_branch() {
     git checkout $current_branch
 }
 
-get_agent_version() {
+get_openhands_version() {
     # IMPORTANT: Because Agent's prompt changes fairly often in the rapidly evolving codebase of OpenHands
     # We need to track the version of Agent in the evaluation to make sure results are comparable
-    AGENT_VERSION=v$(poetry run python -c "import openhands.agenthub; from openhands.controller.agent import Agent; print(Agent.get_cls('$AGENT').VERSION)")
+    OPENHANDS_VERSION=v$(poetry run python -c "import openhands.agenthub; from openhands.controller.agent import Agent; print(Agent.get_cls('$AGENT').VERSION)")
```
Comment (enyst) [inline evaluation/utils/version_control.sh:45]: This still gets agent version though 😄

Comment (neubig) [inline evaluation/utils/version_control.sh:45]: Is this right? It seems like the functionality might not be changed?

Comment (enyst) [reply evaluation/utils/version_control.sh:45]: I think we want this https://github.com/All-Hands-AI/OpenHands/blob/793e142c4a4656543497a8c98935a1558a090a5b/openhands/__init__.py#L6

Comment (xingyaoww) [reply evaluation/utils/version_control.sh]: ahhh sorry forgot to push second commit!

Comment (xingyaoww) [reply evaluation/utils/version_control.sh]: ahhh sorry forgot to push second commit! Should be fixed now

Comment (xingyaoww) [reply evaluation/utils/version_control.sh]: Switched to `get_version` - seems more robust that way

File: evaluation/benchmarks/EDA/scripts/run_infer.sh [hunk 2]
```diff
@@ -36,10 +36,10 @@ fi
 
 # IMPORTANT: Because Agent's prompt changes fairly often in the rapidly evolving codebase of OpenHands
 # We need to track the version of Agent in the evaluation to make sure results are comparable
-AGENT_VERSION=v$(poetry run python -c "import openhands.agenthub; from openhands.controller.agent import Agent; print(Agent.get_cls('$AGENT').VERSION)")
+OPENHANDS_VERSION=v$(poetry run python -c "import openhands.agenthub; from openhands.controller.agent import Agent; print(Agent.get_cls('$AGENT').VERSION)")
```
Comment (neubig) [inline evaluation/benchmarks/EDA/scripts/run_infer.sh:39]: Delete this in favor of `get_openhands_version`?

**Review by enyst — APPROVED** [review]

File: evaluation/benchmarks/EDA/scripts/run_infer.sh (hunk 2 above)
Comment (xingyaoww) [reply evaluation/benchmarks/EDA/scripts/run_infer.sh]: Good catch! Deleted in the latest commit.

**Review by neubig — APPROVED** [review]

**Pull request merged**